File: 2599-make-the-prefix-sum-non-negative/2599-make-the-prefix-sum-non-negative.cpp
class Solution {
public:
    int makePrefSumNonNegative(vector<int>& nums) {
        int res = 0;
        long long sum = 0;
        multiset<int> nags;
        for (auto x:nums) {
            sum += x;
            
            if (x<0) nags.insert(x);
            while (sum<0) {
                sum -= nags.extract(nags.begin()).value();
                res++;
            }
        }
    return res;
    }
};

File: 2599-make-the-prefix-sum-non-negative/2599-make-the-prefix-sum-non-negative.py


class Solution:
    def makePrefSumNonNegative(self, nums: List[int]) -> int:
        h = []
        s = 0
        ans = 0
        for x in nums:
            s += x
            heappush(h, x)
            while s < 0:
                s -= heappop(h)
                ans += 1

        return ans